crypto/s2n_ecc_evp.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "crypto/s2n_ecc.h"
+
+#include <openssl/evp.h>
+
+#include "crypto/s2n_hash.h"
+#include "stuffer/s2n_stuffer.h"
+#include "tls/s2n_tls_parameters.h"
+#include "utils/s2n_safety.h"
+
+#if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0) && !defined(LIBRESSL_VERSION_NUMBER)
+#define S2N_IS_X25519_SUPPORTED 1
+#define S2N_ECC_EVP_SUPPORTED_CURVES_COUNT 3
```

Comment:
We should probably add a comment both here and in the `.c` file reminding people to keep `S2N_ECC_EVP_SUPPORTED_CURVES_COUNT` in sync with the actual array. Otherwise people might modify the array and forget to update this value.